src/Handler/StreamHandler.php

```diff
@@ -304,25 +311,23 @@ private function createStream(RequestInterface $request, array $options)
         }
 
         // Microsoft NTLM authentication only supported with curl handler
-        if (isset($options['auth'])
-            && \is_array($options['auth'])
-            && isset($options['auth'][2])
-            && 'ntlm' == $options['auth'][2]
+        if (isset($options['auth'][2]) && 'ntlm' === $options['auth'][2]
         ) {
             throw new \InvalidArgumentException('Microsoft NTLM authentication only supported with curl handler');
         }
 
         $uri = $this->resolveHost($request, $options);
 
         $context = $this->createResource(
-            function () use ($context, $params) {
+            static function () use ($context, $params) {
                 return \stream_context_create($context, $params);
             }
         );
 
         return $this->createResource(
             function () use ($uri, &$http_response_header, $context, $options) {
-                $resource = \fopen((string) $uri, 'r', null, $context);
+                /** @var resource $resource */
+                $resource = \fopen((string) $uri, 'r', false, $context);
```

Comment:
fopen may return false.. this is hiding a possible error here that it would've be better to keep it inside the baseline.

Note this.. moving towards to 7.x we could have addressed these. For instance this specific thing we could introducing a function under `utils` that could either return a resource of throw an exception.. we wanted to reduce as much as possible the changes from 6.x to 7.x (at least we tried our best)

Reply:
Would throwing an exception if `$resource === false` work?

Reply:
I think your note about introducing a function in `utils` for `7.x` should get addressed in another PR?

Reply:
To be fair... on [this PR ](https://github.com/guzzle/guzzle/pull/2670/files#diff-8196fda03bd99e941e07599792a645a4R70) I also introduced a similar change.. so maybe a runtime exception might not be so bad.. either way if it returns false it will fail down the road..

Reply:
Fixed in 0e8f47a44d10535cedaaee819b9c5c1b575aa291